examples/blasius.py

```diff
@@ -0,0 +1,554 @@
+        logger.info(init_message)
+
+#####################################################################
+
+    def my_write_viz(step, t, dt, state):
+        cv = state.cv
+        dv = state.dv
+        viz_fields = [("CV", cv),
+                      ("dt", dt),
+                      ("DV_U", cv.momentum[0]/cv.mass),
+                      ("DV_V", cv.momentum[1]/cv.mass),
+                      ("DV_Ma", cv.velocity[0]/dv.speed_of_sound),
+                      ("DV_P", dv.pressure),
+                      ("DV_T", dv.temperature)]
+
+        write_visfile(dcoll, viz_fields, visualizer, vizname=vizname,
+                      step=step, t=t, overwrite=True)
+
+    def my_write_restart(step, t, cv):
+        rst_fname = snapshot_pattern.format(cname=casename, step=step, rank=rank)
+        if rst_fname != rst_filename:
+            rst_data = {
+                "local_mesh": local_mesh,
+                "state": cv,
+                "t": t,
+                "step": step,
+                "order": order,
+                "global_nelements": global_nelements,
+                "num_parts": nparts
+            }
+            write_restart_file(actx, rst_data, rst_fname, comm)
+
+#####################################################################
+
+    def my_health_check(cv, dv):
+        health_error = False
+        pressure = force_evaluation(actx, dv.pressure)
+        temperature = force_evaluation(actx, dv.temperature)
+
+        if global_reduce(check_naninf_local(dcoll, "vol", pressure), op="lor"):
+            health_error = True
+            logger.info(f"{rank=}: NANs/Infs in pressure data.")
+
+        if global_reduce(check_naninf_local(dcoll, "vol", temperature), op="lor"):
+            health_error = True
+            logger.info(f"{rank=}: NANs/Infs in temperature data.")
+
+        return health_error
+
+#####################################################################
+
+    def my_pre_step(step, t, dt, state):
+
+        if logmgr:
+            logmgr.tick_before()
+
+        state = force_evaluation(actx, state)
+
+        fluid_state = get_fluid_state(state)
+        dv = fluid_state.dv
+        cv = fluid_state.cv
+
+        if local_dt:
+            t = force_evaluation(actx, t)
+            dt = get_sim_timestep(dcoll, fluid_state, t, dt, current_cfl,
+                gas_model, constant_cfl=constant_cfl, local_dt=local_dt)
+            dt = force_evaluation(actx, actx.np.minimum(dt, current_dt))
+        else:
+            if constant_cfl:
+                dt = get_sim_timestep(dcoll, fluid_state, t, dt, current_cfl,
+                                      t_final, constant_cfl, local_dt)
+
+        try:
+            do_viz = check_step(step=step, interval=nviz)
+            do_restart = check_step(step=step, interval=nrestart)
+            do_health = check_step(step=step, interval=nhealth)
+
+            if do_health:
+                health_errors = global_reduce(my_health_check(cv, dv), op="lor")
```

Comment:
global reduction here will catch when any rank fails health.  inside health is the rank-local check meant to sound off on any rank-specific violation.